### DISCUSSION TITLE
Possible to get hash from url in middleware?

### DISCUSSION THREAD

**Discussion by KolbySisk** [vercel/next.js#38631] [Help]
Hi, I'm using supabase's password reset, which creates a link similar to:
`localhost:3000/#access_token=ABC&type=reset`

I was hoping to use middleware to check for that url and redirect to the appropriate page, but I can't find a way to get the full url in the middleware handler.

`request.url` returns `http://localhost:3000/`
`request.nextUrl` didn't seem to have what I was looking for either.

Is this possible?

**Comment by icyJoseph** [comment]
Could you try stitching it together from:

```js
 const urlObj = new URL(request.nextUrl.pathname, request.nextUrl.origin);
 console.log(urlObj.toString());
```

## Edit

OP wanted the `hash` part of the URL, which is unfortunately not sent by the client.

To be clear, this is not a Next.js quirk, but part of the HTML standard, and all in all, not something browsers do.

**Comment by KolbySisk** [comment]
That also is just `http://localhost:3000/`.

Logging out all of nextUrl I see nothing that contains the hash in the url

**Comment by KolbySisk** [comment]
Aw, according to @timneutkens 

> Fragments (#) are unfortunately never sent to the server. So if you'd do a refresh you would lose the state. There is currently no property available to get the hash.

I assume the same is true of edge functions.

**Reply by icyJoseph** [reply]
oh! I missed that that it was the hash you wanted 😓 , pardon me, but AFAIK those are not sent by the client https://stackoverflow.com/a/3664324

You can check your dev tools network tab, and see that the outgoing requests do not contain the `hash` part of the browser `url`
